lib/spack/spack/cmd/buildcache.py

```diff
@@ -331,12 +333,12 @@ def match_downloaded_specs(pkgs, allow_multiple_matches=False, force=False,
     return specs_from_cli
 
 
-def _createtarball(env, spec_yaml=None, packages=None, add_spec=True,
+def _createtarball(env, spec_file=None, packages=None, add_spec=True,
                    add_deps=True, output_location=os.getcwd(),
                    signing_key=None, force=False, make_relative=False,
                    unsigned=False, allow_root=False, rebuild_index=False):
-    if spec_yaml:
-        with open(spec_yaml, 'r') as fd:
+    if spec_file:
+        with open(spec_file, 'r') as fd:
```

Comment:
This code-block needs to be updated for backwards compatibility, it has a hardcoded read from yaml and uses yaml-specific filenames.

Probably also needs a test for this, since we got through all the tests without tripping over it. I think we can test it by installing a package, manually removing the spec.json and writing a spec.yaml, and then creating a buildcache from the package.

Reply:
Good catch thanks. First, I confirmed that indeed this could be yaml if the user has yaml-based installations hanging around. Second, I have added in a check for yaml/json. I see the `buildcache create` test in `test/cmd/buildcache.py`, but I am looking to see where `buildcache install` gets tested... 
A simple test would just be to do a `create` followed by an `install`...